pywbem/_logging.py

```diff
@@ -32,157 +34,206 @@
+  subsequently created WBEM connections.
+
+  Because the pywbem loggers can generate large quantities of information, the
+  user can control the quantity of information produced by each pywbem logger
+  by setting a detail level for each logger when activating a WBEM connection
+  for logging.
+
+  Activation and setting detail levels are pywbem features so it requires using
+  the functions described in :ref:`Logging configuration functions`.
+
+
+.. _`Logging configuration functions`:
+
+Logging configuration functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The pywbem loggers may be configured and/or WBEM connections may be activated
+for logging through the following pywbem functions.
+
+These functions are the only mechanism for setting the detail level of a pywbem
+logger and for activating WBEM connection(s) for logging.
+
+* :meth:`~pywbem.WBEMConnection.configure_logger` - Configure the pywbem
+  loggers and optionally activate WBEM connections for logging and setting a
+  log detail level.
+
+* :func:`configure_loggers_from_string` - Configure the pywbem loggers and
+  optionally activate WBEM connections for logging and setting a log detail
+  level, from a log configuration string.
+
+  This is most useful in defining the pywbem logging from a command line tool
+  such as pywbemcli so the complete logger configuration can be compressed into
+  a single command line string.
+
 
 .. _`Logging configuration examples`:
 
 Logging configuration examples
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Configure and activate loggers with
-:meth:`~pywbem.WBEMConnection.configure_logger`.
+Examples for using :meth:`~pywbem.WBEMConnection.configure_logger` for
+configuring pywbem loggers and for activating WBEM connections for logging:
 
-* Configure the "pywbem.api" logger for summary information output to a file
-  and activate that logger for all subsequently created
-  :class:`~pywbem.WBEMConnection` objects::
+* Example: Configure the `'pywbem.api'` logger with detail level `'summary'`
+  and output to stderr, and activate all subsequently created WBEM connections
+  for logging::
 
-    WBEMConnection.configure_logger('api', dest='stderr',
+    WBEMConnection.configure_logger('api',
+                                    log_dest='stderr',
                                     detail_level='summary',
                                     connection=True)
-    conn = WBEMConnection()  # logs WBEMConnection api calls and responses
 
-* Configure and activate a single :class:`~pywbem.WBEMConnection` object logger
-  for output of summary information for both "pywbem.api" and "pywbem.http"::
+    # All of the following connections will log:
+    conn1 = WBEMConnection(...)
+    conn2 = WBEMConnection(...)
+
+* Example: Configure all pywbem loggers with the default detail level (`'all'`)
+  and output to a file, and activate a single existing WBEM connection for
+  logging::
 
     conn = WBEMConnection()
```

Comment:
Put a url in this or the ... to indicate parameters.

Reply:
DONE